sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -401,17 +394,17 @@ namespace Azure { namespace Core { namespace Http {
       this->m_connection = GetCurlConnection(this->m_request);
       this->m_bodyStartInBuffer = -1;
       this->m_innerBufferSize = Details::c_DefaultLibcurlReaderSize;
-      this->m_rawResponseEOF = false;
       this->m_isChunkedResponseType = false;
       this->m_uploadedBytes = 0;
+      this->m_sessionTotalRead = 0;
     }
 
     ~CurlSession() override
     {
       // mark connection as reusable only if entire response was read
       // If not, connection can't be reused because next Read will start from what it is currently
       // in the wire. We leave the connection blocked until Server closes the connection
-      if (this->m_rawResponseEOF)
+      if (IsEOF())
```

Comment:
Is this really the best thing to do? Maybe we should read to the end and then put the connection back in the pool. Is there ANY benefit to just letting the connection linger?

Reply:
it's gonna take some time if the body to read is huge. In this case we might as well close this connection.

Reply:
yeah, we can't read to end as @JinmingHu-MSFT  said.

Note: We **don't** let connection linger.  I will update the comment on it.
As soon as the connection is not moved back to the pool, it is destroyed.  It's destructor will clean libcurl handle, so the connection is closed.